python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+        # First assume dipole is not separated by more than 5*psfSigma.
+        maxSep = self.psfSigma * maxSepInSigma
+
+        # Note - this may be a cheat to assume the dipole is centered in center of the footprint.
+        if np.sum(np.sqrt((np.array(cenPos) - fpCentroid)**2.)) > maxSep:
+            cenPos = fpCentroid
+        if np.sum(np.sqrt((np.array(cenNeg) - fpCentroid)**2.)) > maxSep:
+            cenPos = fpCentroid
+
+        # parameter hints/constraints: https://lmfit.github.io/lmfit-py/model.html#model-param-hints-section
+        # might make sense to not use bounds -- see http://lmfit.github.io/lmfit-py/bounds.html
+        # also see this discussion -- https://github.com/scipy/scipy/issues/3129
+        gmod.set_param_hint('xcenPos', value=cenPos[0],
+                            min=cenPos[0]-maxSep, max=cenPos[0]+maxSep)
+        gmod.set_param_hint('ycenPos', value=cenPos[1],
+                            min=cenPos[1]-maxSep, max=cenPos[1]+maxSep)
+        gmod.set_param_hint('xcenNeg', value=cenNeg[0],
+                            min=cenNeg[0]-maxSep, max=cenNeg[0]+maxSep)
+        gmod.set_param_hint('ycenNeg', value=cenNeg[1],
+                            min=cenNeg[1]-maxSep, max=cenNeg[1]+maxSep)
+
+        # Estimate starting flux. This strongly affects runtime performance so we want to make it close.
+        if False:
+            # Value to convert peak value to total flux based on flux within psf
+            psfImg = self.diffim.getPsf().computeImage()
+            pkToFlux = np.nansum(psfImg.getArray()) / self.diffim.getPsf().computePeak()
+
+            bg = np.nanmedian(diArr)  # Compute the dipole background (probably very close to zero)
+            startingPk = np.nanmax(diArr) - bg   # use the dipole peak for an estimate.
+            posFlux, negFlux = startingPk * pkToFlux, -startingPk * pkToFlux
+
+            if len(pks) >= 1:
+                posFlux = pks[0].getPeakValue() * pkToFlux
+            if len(pks) >= 2:
+                negFlux = pks[1].getPeakValue() * pkToFlux
+
+        # Use the (flux under the dipole)*5 for an estimate.
+        # Lots of testing showed that having startingFlux be too high was better than too low.
+        startingFlux = np.nansum(np.abs(diArr) - np.nanmedian(np.abs(diArr))) * 5.
+        posFlux = negFlux = startingFlux
+
+        # TBD: set max. flux limit?
+        gmod.set_param_hint('flux', value=posFlux, min=0.1)
+
+        if separateNegParams:
+            # TBD: set max negative lobe flux limit?
+            gmod.set_param_hint('fluxNeg', value=np.abs(negFlux), min=0.1)
+
+        # Fixed parameters (dont fit for them if there are no pre-sub images or no gradient fit requested):
+        # Right now, we use the linear model to fit the background and initialize the parameters.
+        # It might be better to just subtract the linear model from the data and then don't fit the background
+        # again.
```

Comment:
No ticket necessary. Since I haven't tested the fitter on real, messy data yet, I want to include this alternative which may be a fallback in difficult situations. I have updated the comment accordingly.